cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/votable/tabledata/TableDataElementWriter.java

```diff
@@ -105,30 +105,47 @@ public void write(Writer out) throws IOException {
         log.debug("Writing TABLEDATA element - starting");
         out.write("<TABLEDATA>");
 
-        long rowCount = 1;
+        long rowCount = 0;
+        boolean success;
 
         while (rowIter.hasNext()) {
+            rowCount ++;
             List<Object> row = rowIter.next();
-            writeRow(out, row);
+            success = writeRow(out, row);
```

Comment:
I think the structure used in the BinaryElementWriter, where `writeRow(...)` just throws whatever exception occurs and the try/catch (set trailer and break) is in the while loop is simpler and works for all failures.

Please update the TableDataElementWriter to the same structure.